fe/fe-core/src/main/java/com/starrocks/lake/LakeAggregator.java

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.lake;
+
+import com.starrocks.common.StarRocksException;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.server.WarehouseManager;
+import com.starrocks.system.ComputeNode;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+
+public class LakeAggregator {
+    private static final Logger LOG = LogManager.getLogger(Utils.class);
+
+    public LakeAggregator() {}
+
+    // TODO(zhangqiang)
+    // Optimize the aggregator selection strategy
+    public static ComputeNode chooseAggregatorNode(long warehouseId) {
+        try {
+            WarehouseManager warehouseManager = GlobalStateMgr.getCurrentState().getWarehouseMgr();
+            List<ComputeNode> candidateNodes = warehouseManager.getAliveComputeNodes(warehouseId);
+            if (candidateNodes != null && !candidateNodes.isEmpty()) {
+                return candidateNodes.get(ThreadLocalRandom.current().nextInt(candidateNodes.size()));
+            } else {
+                Long nodeId = GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo()
```

Comment:
Why is this code needed when the warehouseId is specified? This meaningless downgrade solution will cause troubles in future use.Why not just report an error?

Reply:
> Why is this code needed when the warehouseId is specified? This meaningless downgrade solution will cause troubles in future use.Why not just report an error?

The publish can not failed in our system design, so we should try the best to finish publish. If we do not support downgrade, if the warehouse is dropped before publish, the publish will always failed and the ingestion will be blocked.